### PR TITLE
Update HPKE dependency and ODoH version.

### LINKED ISSUE #9: Update to version -05
[Version-05](https://tools.ietf.org/html/draft-pauly-dprive-oblivious-doh-05) now exists. Let's move to it!

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #9.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,4 +2,8 @@ module github.com/cloudflare/odoh-go
 
 go 1.14
 
-require github.com/cisco/go-hpke v0.0.0-20201215202025-9cebdf8f33d4
+require (
+	github.com/cisco/go-hpke v0.0.0-20210215210317-01c430f1f302
+	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83 // indirect
```
Comment (tanyav2) [inline go.mod:7]: Please run `go mod tidy` before committing changes to get rid of indirect dependencies

**Review by tanyav2 — CHANGES_REQUESTED** [review]
This doesn't update the test vectors and the draft version in the `README`.

File: go.mod (hunk 1 above)
Comment (chris-wood) [reply go.mod:7]: Done, but the indirect dependencies persist.

**Comment by chris-wood** [comment]
@tanyav2 please have another look and merge if there are no more blocking changes!

File: go.mod (hunk 1 above)
Comment (tanyav2) [reply go.mod:7]: That's okay, you could just delete them manually, but it doesn't really matter in any case.

**Review by tanyav2 — APPROVED** [review]

**Pull request merged**